### DISCUSSION TITLE
I do not understand why git clone does not work on other computers

### DISCUSSION THREAD

**Discussion by DanielCruz-fs** [community/community#50882] [Programming Help]
I create a repo on my windows pc, then i wanted to continue from my old linux ubuntu but i can not get
the git clone work.... it says does not support for password etc... Uhhhhggg this is so frustrating can
someone please give me a link to an easy solution i looked so many tuts but  i just cannot get them

**Comment by jge162** [comment]
You can use your token. Have you tried that.

**Comment by neutrinomuon** [comment]
Hi!

Oh, yes because now the access is only supported through tokens, the password access was disabled. So, you need to create one in order to access your private repositories or 'push' something new.

1. Go to your GitHub account and click on 'Settings';
2. Then, click on "<> Developer settings" on the bottom left list of options;
3. Click on "Personal access tokens" and then click on "Tokens (Classic)". You can choose another option, but I find this easier;
4. You can choose your token's name, select the appropriate permissions for your use and how long it will be valid. However, make sure to select "repo" to access your private repositories;
5. Then you just new to choose "Generate new token (Classic)" - For general use;
6. Don't forget to copy the token to your clipboard and save it;

In the Ubuntu terminal, navigate to the directory where you want to clone the repository. Now it is time to run the git clone command. For example, if the repository URL is https://github.com/username/repo.git and your personal access token is abcdefg123456789, just do:

<pre>
git clone https://username:abcdefg123456789@github.com/username/repo.git
</pre>

I hope this helps. Please let me know if it works.

**Comment by sergeyleschev** [comment]
Try using a personal access token instead of password authentication for cloning the repository.

**Comment by vitalNohj** [comment]
I'm having the same problem. I can create a private repo on Github, use Github Desktop to clone the repo on my desktop without any issues, even without authentication. But when I use my laptop with Ubuntu and Github Desktop, it requires a personal access token. I have tried CLI and its the same problem, it will not accept my credentials!

The private repo is setup under an organization. I've setup multiple organizations and never had this problem before. The org is accepting my access token too.

**Comment by xynapz** [comment]
I had a same problem some time ago, but i fixed it by setting up ssh with git. Here is the solution I used.

# GIT SSH setup in ubuntu.

 1. Install the openssh-client if it is not already installed, and of course git.
    ```
    sudo apt update && sudo apt install -y openssh-client git
    ```
 
 2. Create user's ssh directory and a sub directory where your dedicated GitHub ssh key will be stored.
    ```
    mkdir -p ~/.ssh/github
    chmod 700 ~/.ssh ~/.ssh/github
    ```

 3. Generate the SSH key (the output key will have octal permissions 600).
    ```
    ssh-keygen -t rsa -b 4096 -C 'your@email.com' -f ~/.ssh/github/id_rsa -q -N ''
    ```
    * ```-q - silence ssh-keygen; -N '' - empty (without) passphrase, you can assign one if you want.```

 4. Copy the content of the file id_rsa.pub, use the following command to output it.

    ```
    cat ~/.ssh/github/id_rsa.pub
    ```

 5. Go to your GitHub account. From the drop-down menu in upper right corner select Your profile. Click on Edit profile button and then select SSH and GPG keys. Click on the New SSH Key button. Type some meningful for a Title and paste the content of ~/.ssh/github/id_rsa.pub in the field Key. Then click on the Add SSH Key button.

 6. Create the ~/.ssh/config file, if it doesn't already exist.
    ```
    touch ~/.ssh/config
    chmod 600 ~/.ssh/config
    ```

 7. Edit the config file and add the following entry for the new SSH key.
    ```
    Host github.com
    IdentityFile ~/.ssh/github/id_rsa
    ```

 8. Test the setup. Use the following command.
    ```
    ssh -T git@github.com
    ```
[Reference][reference]

[reference]: https://askubuntu.com/questions/1097038/how-do-i-setup-ssh-key-based-authentication-for-github-by-using-ssh-config-fi
